scene/2d/skeleton_2d.cpp

```diff
@@ -325,7 +325,7 @@ bool Bone2D::_editor_get_bone_shape(Vector<Vector2> *p_shape, Vector<Vector2> *p
 		rel = rel.rotated(-get_global_rotation()); // Undo Bone2D node's rotation so its drawn correctly regardless of the node's rotation
 	} else {
 		real_t angle_to_use = get_rotation() + bone_angle;
-		rel = Vector2(cos(angle_to_use), sin(angle_to_use)) * (length * MIN(get_global_scale().x, get_global_scale().y));
+		rel = Vector2(cos(angle_to_use), sin(angle_to_use)) * length * get_global_scale().abs();
```

Comment:
It seems that the shape is draw AFTER the node draw itself.  

Which means that shape is not affected by the absolute value of the node scale, only by the sign. So we don't need the signs that's why is removed with `abs()`:

[Screencast from 2024-01-10 16-00-02.webm](https://github.com/godotengine/godot/assets/9352894/49f8a438-1f19-4da6-8c33-5253051cc3de)

This may also be the reason for SkeletonModificationLookAt not pointing to the right location and Y is negative:  

[Screencast from 2024-01-10 16-01-43.webm](https://github.com/godotengine/godot/assets/9352894/c5c1eab4-5946-429b-af88-f020dccdde97)

**Note**: `get_global_scale` always give a positve X, this is why scaling X doesn't show us any problem (https://docs.godotengine.org/en/stable/classes/class_node2d.html#class-node2d-property-scale).

Reply:
The flip in -Y probably involves a normalization problem. It may be a specification. Do you have any ideas in this part @reduz? Since I remember having a similar discussion before in https://github.com/godotengine/godot/pull/69202.

However, at least in SkeletonModificationLookAt calculation, it should be resolved in SkeletonModificationLookAt, and it should not be correct way to force abs() to fix it here.

Reply:
> However, at least in SkeletonModificationLookAt calculation, it should be resolved in SkeletonModificationLookAt, and it should not be correct way to force abs() to fix it here.

I agree, but i'm not sure if is SkeletonModificationLookAt fault in this case.  
Only the bone draw seems to be wrong because node and child nodes are actually looking at the at target.

[Screencast from 2024-01-10 17-17-19.webm](https://github.com/godotengine/godot/assets/9352894/35833bc1-6106-4acc-a3d7-a7ee6e0d35ba)

Reply:
Skeleton2DGizmo doesn't exist in Godot (only Skeleton3DGizmo).  
skeleton_2d_editor_plugin.cpp only seems to have the logic to rest & overwrite rest pose.  

skeleton_2d.cpp comment indicates that we are looking at the right file:  
https://github.com/godotengine/godot/blob/26b1fd0d842fa3c2f090ead47e8ea7cd2d6515e1/scene/2d/skeleton_2d.cpp#L191-L192  

I believe that this PR is in the right place for the gizmo drawing logic.

Reply:
You should see `Polygon2DEditor`.

Reply:
I looked into it, and I think the gizmo direction mismatch from axes is a specification. Even if there is no negative scale, the BoneAngle will generate a visual offset of the gizmo.

![image](https://github.com/godotengine/godot/assets/61938263/2aa1632f-83c1-4918-95db-41aa92551135)

Finally, the LookAt issue is that it just doesn't take that offset into account.

Reply:
> You should see `Polygon2DEditor`.

Still need to see :laughing: 

> I looked into it, and I think the gizmo direction mismatch from axes is a specification. Even if there is no negative scale, the BoneAngle will generate a visual offset of the gizmo.

Sorry, I failed understand your point.  
In my case I'm always using BoneAngle 0º, so the expected is to match X-axis.  
I do rotate the Node rotation, but when using BoneAngle 0º this would means matching X-axis. Right?  

> In other words, if we force Polygon2DEditor to draw Gizmo on X-axis, Bone Angle will not work.

What you mean by forcing it to draw Gizmo on X-axis?  

---

What I understand is right now is:
- Gizmo is drawn every frame
- Gizmo is not a child of Bone2D so it's not affected by it scales
- We need to add scale when drawing
- Right now we are getting the lower scale
  - Which brings the weird behavior when one scale is negative
- We use absolute to calculate the correct the length
  - `Vector2(cos(angle_to_use), sin(angle_to_use)) * length * get_global_scale().abs()`
    - `Vector2(cos(angle_to_use), sin(angle_to_use))` is the vector direction
    - `length * get_global_scale().abs()` is the length scaled (a number)
      - If we do **not** use the absolute here
        - `length * get_global_scale()` this will give us a vector and I have no idea why we would multiple direction by a vector.

Correct me if I'm seeing something wrong.

Reply:
> In my case I'm always using BoneAngle 0º, so the expected is to match X-axis.

That is, if BoneAngle is zero and the scale is isotropic.

I am not so familiar with the Bone2D specification, but it seems that if the scales are not isotropic, they are implicitly representing the BoneAngle.

The fundamental problem is that Gizmo2D cannot have distorted axes like 3D, but if BoneAngle is non-zero, even if it is due to non-isotropic scales, the bone is by definition not aligned with the X axis, so the offset should not be corrected by abs().

Reply:
I don't understand enough about the subject... I need to study  

![sad gif - konosuba](https://c.tenor.com/m7zC_iHi-r0AAAAC/tenor.gif)

Reply:
Oh, sorry. I looked at the more details and may have misunderstood. It looked like the anisotropic Scale was acting strangely, but in fact it may just be that LookAt was ignoring the Auto calculation setting and setting the BoneAngle. Well, in any case, I believe there is a bug in LookAt.

Reply:
Seems Bone Angle has a different issue and seems to be due to the fact that Auto Calculate Length and Angle is unstable. I send PR for that https://github.com/godotengine/godot/issues/91759.

Reply:
After further investigation, I see that the abs() is necessary here because there is reverse rotation, but I think the original implementation which needs reverse rotation was odd to begin with.

I have reviewed the implementation in https://github.com/godotengine/godot/pull/91731, so would appreciate your testing. We shouldn't need abs() there.

Reply:
I did my best to find some error, but it seems to be okay.